src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -206,6 +209,28 @@
       OutputListLocation="$(OutDir)assemblylist.txt"
       Condition="'$(UseLegacyXunitPerfRunner)'=='true'"
      />
+     
+     <!-- For Desktop Tests, we need to generate Binding Redirects. -->
+     <ItemGroup>
+        <_FileNamesInReferenceCopyLocalPaths Include="@(ReferenceCopyLocalPaths->'%(Filename)%(Extension)')">
+          <SourcePath>%(ReferenceCopyLocalPaths.Identity)</SourcePath>
+        </_FileNamesInReferenceCopyLocalPaths>
+        <_FileNamesInTestCopyLocalByFileNameWithoutDuplicates Include="@(_TestCopyLocalByFileNameWithoutDuplicates->'%(Filename)%(Extension)')">
+          <SourcePath>%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)</SourcePath>
+        </_FileNamesInTestCopyLocalByFileNameWithoutDuplicates>
+        <_FilteredListForBindingRedirect Include="@(_FileNamesInReferenceCopyLocalPaths)" Exclude="@(_FileNamesInTestCopyLocalByFileNameWithoutDuplicates)" />
+        <_FilteredListForBindingRedirect Include="@(_FileNamesInReferenceCopyLocalPaths)" />
+        <_FilteredAssembliesForBindingRedirect Include="%(_FilteredListForBindingRedirect.SourcePath)" Condition="$([System.String]::Copy('%(_FilteredListForBindingRedirect.SourcePath)').EndsWith('.dll')) or $([System.String]::Copy('%(Remaining.SourcePath)').EndsWith('.exe'))" />
+        <ExecutablesForBindingRedirect Include="%(IncludedFileForRunnerScript.SourcePath)" Condition="$([System.String]::Copy('%(IncludedFileForRunnerScript.SourcePath)').EndsWith('.exe'))"/>
```

Comment:
Suggestion, instead of all these item manipulation expressions, is it possible to just give @(ReferenceCopyLocalPaths) to the task ? It can filter out all but exe and dll, and can consult %(SourcePath) if needed.